### PR TITLE
Forms

### PR DESCRIPTION
**Pull request by HSMathebula**
You can find the features added for this Pull Request below.

- A method called current_user was created in your ApplicationController to make current user data available to all controllers. It returns the first user from the database.
- Forms were created to perform the following functions:
  - Creates a Post for the current_user.
  - Creates a comment for the current_user.
- Methods and views are added to Users to add likes to Posts.
- After saving the form, updates the posts' likes and comments counter.

### CODE REVIEW

File: config/routes.rb [hunk 1]
```diff
@@ -3,4 +3,9 @@
   resources :users, only: [:index, :show] do
     resources :posts, only: [:index, :show]
   end
+  get "/posts/new", to: "posts#new"
+  post "posts/new", to: "posts#create"
```
Comment (codepantha) [inline config/routes.rb]: - [ ] Please modify your routes to take into account that a new post belongs to a user. An easy way would be to include the new and create actions in the resource routes you which you already have.

File: config/routes.rb [hunk 2]
```diff
@@ -3,4 +3,9 @@
...
+  get '/comments/new', to: "comments#new"
+  post '/comments/new', to: "comments#create"
```
Comment (codepantha) [inline config/routes.rb]: - [ ] Please modify your routes to take into account that comments belong to a post. 
You should have something like `get '/posts/:post_id/comments/new`. Same goes for the post as well.

File: app/controllers/comments_controller.rb [hunk 3]
```diff
@@ -0,0 +1,15 @@
+class CommentsController < ApplicationController
+  def new
+    @comment = Comment.new
+  end
```
Comment (codepantha) [inline app/controllers/comments_controller.rb:4]: - [ ] Please create a views/comments/new.html.erb file which will have a form that accepts input for creating comments.

File: config/routes.rb [hunk 4]
```diff
@@ -3,4 +3,9 @@
...
+  post '/likes/new', to: "likes#create"
```
Comment (codepantha) [inline config/routes.rb]: - [ ] Please modify your routes to take into account that a like belongs to a given post.
You should have something like `post '/posts/:post_id/likes`.

**Review by codepantha — CHANGES_REQUESTED** [review]
# STATUS: REQUIRES CHANGES ♻️ 

## Hi @HSMathebula 👋, 

Good job so far!
You still need to work on some issues to go to the next project, but you are almost there!

## Highlights 🎯 
✔️ Descriptive PR
✔️ Used correct GitFlow
✔️ No linter errors

## Required Changes ♻️
_Check the comments under the review._

## Optional suggestions
_Every comment with the [OPTIONAL] prefix is not crucial enough to stop the approval of this PR. However, I strongly recommend you to take them into account as they can make your code better._

You can also consider:
- n/a

Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag me in your question so I can receive the notification.

**_Please, do not open a new Pull Request for re-reviews. You should use the same Pull Request submitted for the first review, either valid or invalid unless it is requested otherwise._**

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561) you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Review by CollinsTatang — APPROVED** [review]
# Approved 👏  
### **Highlights 🚀 :heavy_check_mark: :heavy_check_mark:** 

- The repo looks professional.:heavy_check_mark:
- Git/GitHub flow best practices have been followed:heavy_check_mark:
- You have descriptive PR:heavy_check_mark:

Your project is complete!  There is nothing else to say other than... it's time to merge it :shipit: 
### **Congratulations!** 🎉 
image



Cheers and Happy coding!👏👏👏

Feel free to leave any questions or comments in the PR thread if something is not 100% clear.
Please, remember to tag @CollinsTatang in your question so I can receive the notification.

------
_As described in the [Code reviews limits policy](https://microverse.zendesk.com/hc/en-us/articles/1500004088561)  you have a limited number of reviews per project (check the exact number in your Dashboard). If you think that the code review was not fair, you can request a second opinion using [this form](https://airtable.com/shrQAqnBwek5a0O0s)._

**Pull request merged**